### ISSUE TITLE
Allow to configure the base for Integer parsing

### ISSUE THREAD

**Issue by ccremer** [urfave/cli#1462]
## Checklist

* [x] Are you running the latest v2 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v2 manual is [here](https://github.com/urfave/cli/blob/main/docs/v2/manual.md).
* [x] Did you perform a search about this feature? Here's the [Github guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## What problem does this solve?

We pass a year and month as CLI flags/envs (each have their own flag/env var).
Unfortunately, when passing a month with preceding 0 like `08` for August it gets parsed as octal, because of `strconv.ParseInt(..., 0, ...) (see https://github.com/urfave/cli/blob/b4df361387226948b6cbaf53da539e26c7562cce/flag_int.go#L47) it's set to autodetect (https://pkg.go.dev/strconv#ParseInt).
In cases where the flag or env var is set by another tool with preceding 0, this can lead to unexpected issues like https://github.com/vshn/appuio-odoo-adapter/issues/57.

If we could configure which base the parser should use, it would solve our issue.
See playground here: https://go.dev/play/p/a8VllHp_Z6D

## Solution description

Add a property to all Int flags to let us explicitly configure the base for our parsing, e.g. `Base` or `ParseBase` (name tbd) that is then used in `ParseInt`.
Since Go's default value for Integers are 0, it would match what is currently used in `ParseInt`, thus I don't think it causes a breaking change.

## Describe alternatives you've considered

* Configure the month flag as StringFlag and do the parsing explicitly within the Before/Action functions.

Other workarounds are all outside of urface/cli:

* Combine the year and month flag into a date flag with format ("yyyy-mm"), this would be a breaking change for the user.
* Force any external tool that provide the value for the month to trim zeros when invoking the CLI. Depending on the tool used this might not always be possible.

**Comment by davidgubler** [comment]
In my opinion the default should be base 10 parsing and only base 10 parsing, as that is what a user would expect. Having such magic enabled by default causes unexpected reliability issues. Note that, in our case of parsing months, '01' through to '07' parse correctly, as do '10', '11' and '12'; the issue in this case is very subtle as it only applies to '08' and '09'. This is exactly why this kind of magic should not be the default.

Now I realize that changing this default behavior is a breaking change, but I think that this is something worth addressing in a major version in order to solve this problem once and for all for future users.

**Comment by dearchap** [comment]
@ccremer Have you looked at using the cli.TimestampFlag for this purpose, you could do something like

```
--time 2006-01-02T15:04:05Z
```

**Comment by ccremer** [comment]
@dearchap , yes I'm aware, I've listed it as a workaround in the alternatives already.

**Comment by dearchap** [comment]
@ccremer Would you like to submit a PR for this ? Theoretically you could have your own implementation of a different base using GenericFlag. Adding it to all Int Flag types is a lot of work for us right now.

**Comment by ccremer** [comment]
@dearchap I can try. I might be able to find a bit of time this week.

**Comment by ccremer** [comment]
@dearchap I've opened #1464 as a start.
